### PR TITLE
workload/tpcc: extend --wait to also support fractions

### PR DESCRIPTION
**Pull request by knz**
In order to use TPC-C for operational testing (i.e. NOT benchmarking)
it is useful to increase the rate of transactions without
altogether saturating the TPS capacity (`--wait=false`).

For this purpose, this commit extends the syntax accepted by `--wait`
to recognize a multiplier which is applied to all wait times. The
default is 1.0 (i.e. 100% of the wait time required by a benchmark).
The words `true`/`on` and `false`/`off` are aliases for 1.0 and 0.0,
respectively

### CODE REVIEW

**Review by danhhz — COMMENTED** [review]
I'm okay with this, though we should probably deprecate the --waits flag since --wait-fraction=0 should be the same thing, right? Mind running this by nathan or awerner to see if they have thoughts on whether tpcc is the right thing for what you'd looking for?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/39685#-:-LmLWQBt7U-ShyuWpenn:b-kj9545)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @danhhz and @tbg)

**Comment by knz** [comment]
I have talked to @nvanbenschoten and we agreed that this is an OK way to go forward, as long as the users of the flag are mindful that it changes the amount of contention.

@danhhz what is it that you'd like me to do about `--waits`? Are you not concerned about making a flag disappear outright? Should I mark it as hidden/deprecated instead?

**Comment by petermattis** [comment]
You could change the existing `--waits` flag to take a fractional argument. So `--waits=true` would be equivalent to `--waits=1` and `--waits=false` would be equivalent to `--waits=0`. But then you could also specify `--waits=0.1`. Perhaps overly clever, but that's what I'm here for.

**Comment by danhhz** [comment]
I was imagining adding a deprecation notice to the --waits flag description. If that makes 19.2, I'd be comfortable removing the flag in 20.1

**Comment by knz** [comment]
> You could change the existing --waits flag to take a fractional argument. So --waits=true would be equivalent to --waits=1 and --waits=false would be equivalent to --waits=0. But then you could also specify --waits=0.1. Perhaps overly clever, but that's what I'm here for.

I understand how to do this. I think it is a good idea because it's upward compatible and doesn't require a deprecation cycle. I'll try that.

**Comment by knz** [comment]
Modified as suggested, RFAL

**Review by danhhz — APPROVED** [review]
this is great, much better than what I was suggesting. :lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/39685#-:-Lnro7w59TWR_nzcMnKz:b-b5rq2c)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @knz and @tbg)

___
*[pkg/workload/tpcc/tpcc.go, line 157 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/39685#-LnroGag42pxSn2P5do0:-LnroGag42pxSn2P5do1:boqjte1) ([raw file](https://github.com/cockroachdb/cockroach/blob/3ce10d47566fc357e14316f2e105705a64258752/pkg/workload/tpcc/tpcc.go#L157)):*
> ```Go
> 			`split`:              {RuntimeOnly: true},
> 			`wait`:               {RuntimeOnly: true},
> 			`wait-fraction`:      {RuntimeOnly: true},
> ```

no longer necessary, right?

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/39685)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @danhhz and @tbg)

___
*[pkg/workload/tpcc/tpcc.go, line 157 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/39685#-LnroGag42pxSn2P5do0:-LnrpOtc6Pn1_enOZ9z2:b-9vhyeg) ([raw file](https://github.com/cockroachdb/cockroach/blob/3ce10d47566fc357e14316f2e105705a64258752/pkg/workload/tpcc/tpcc.go#L157)):*
<details><summary><i>Previously, danhhz (Daniel Harrison) wrote…</i></summary><blockquote>

no longer necessary, right?
</blockquote></details>

Indeed, fixed.

**Comment by knz** [comment]
bors r=danhhz,petermattis

**Comment by knz** [comment]
there was a lint failure, sorry about that

bors r=danhhz,petermattis

File: pkg/workload/tpcc/tpcc.go [hunk 1]
```diff
@@ -200,15 +241,15 @@ func (w *tpcc) Hooks() workload.Hooks {
 				// waiting, we only use up to a set number of connections per warehouse.
 				// This isn't mandated by the spec, but opening a connection per worker
 				// when they each spend most of their time waiting is wasteful.
-				if !w.doWaits {
+				if w.waitFraction == 0 {
 					w.numConns = w.workers
 				} else {
 					w.numConns = w.activeWarehouses * numConnsPerWarehouse
```
Comment (nvanbenschoten) [inline pkg/workload/tpcc/tpcc.go]: Do we need to scale the number of connections by the wait fraction? I'd expect them to saturate as-is with a fraction like `0.1`.

**Pull request merged**